flight/PiOS/Common/pios_esctelemetry.c

```diff
@@ -0,0 +1,244 @@
+/**
```

Comment:
This design will not work for multiple different ESC protocols? This should be `PIOS_DShot_Telemetry` or something? And other ESC telemetry drivers need a way to hook in somehow. Other protocols existed before DShot, and we have already discussed supporting them, so a generic `PIOS_ESCTelemetry` implentation that can hook into multiple different backend drivers is essential IMO.

Reply:
This round gonna do it just DShot specific and rename the current driver accordingly. Later on we could revisit this and split things into generic parts and actuator specific telemetry. Haven't thought it through where to draw the line yet.